src/native/libs/build-native.sh

```diff
@@ -42,11 +42,14 @@ __VerboseBuild=false
 source "$__RepoRootDir"/eng/native/build-commons.sh
 
 # Set cross build
-
-if [[ "$__TargetArch" == wasm ]]; then
+if [[ "$__TargetOS" == Browser ]]; then
```

Comment:
we probably also don't want the default __CMakeArgs stuff for WASI in the `else` so we need to add a no-op `elif` like we do for iOS/tvOS